typescript/cli/src/context/strategies/signer/MultiProtocolSignerManager.ts

```diff
@@ -123,30 +124,57 @@ export class MultiProtocolSignerManager {
   ): Promise<{ chain: ChainName } & SignerConfig> {
     const { protocol } = this.multiProvider.getChainMetadata(chain);
 
-    // For Cosmos, we must use strategy config
+    let config = await this.extractPrivateKey(chain);
+
+    // For Cosmos, we get additional params
     if (protocol === ProtocolType.CosmosNative) {
-      return this.resolveCosmosNativeConfig(chain, this.options.key);
+      const provider =
+        await this.multiProtocolProvider.getCosmJsNativeProvider(chain);
+      const { bech32Prefix, gasPrice } =
+        this.multiProvider.getChainMetadata(chain);
+
+      config = {
+        ...config,
+        extraParams: { provider, prefix: bech32Prefix, gasPrice },
+      };
     }
 
-    // For other protocols, try CLI/ENV keys first, then fallback to strategy
-    const config = await this.extractPrivateKey(chain);
     return { chain, ...config };
   }
 
   /**
    * @notice Gets private key from strategy
    */
   private async extractPrivateKey(chain: ChainName): Promise<SignerConfig> {
-    if (this.options.key) {
+    if (this.options.key && typeof this.options.key === 'object') {
+      assert(
+        this.options.key[chain],
+        `Key flag --key.${chain} for chain ${chain} not provided`,
+      );
       this.logger.debug(
-        `Using private key passed via CLI --key flag for chain ${chain}`,
+        `Using private key passed via CLI --key.${chain} flag for chain ${chain}`,
```

Comment:
also concur that this should be --key.protocol instead of chain mostly because currently --key assumes all chains are ethereum.

Reply:
Alright will rework this PR so that key inputs are per protocol now